app/react/Metadata/components/GroupedGeolocationViewer.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import GeolocationViewer from 'app/Metadata/components/GeolocationViewer';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+import { Translate } from 'app/I18N';
+import { IStore } from '../../istore';
+import { ensure } from '../../../shared/tsUtils';
+import { Pill } from './Pill';
+
+interface GeolocationMarker {
+  lat: string;
+  lon: string;
+  label?: string;
+  color?: string;
+}
+
+interface GroupMember {
+  label: string;
+  name: string;
+  translateContext: string;
+  value: GeolocationMarker[];
+}
+
+interface TemplatesInfo {
+  [template: string]: {
+    name: string;
+    color: string;
+  };
+}
+
+export interface GroupedGeolocationViewerProps {
+  members: GroupMember[];
+  templatesInfo: TemplatesInfo;
+}
+
+const notLabeledOrMultiple = (member: GroupMember) =>
+  member.value.length === 1 && !member.value[0].label;
+
+const getFirstGroupInfo = (members: GroupMember[], templatesInfo: TemplatesInfo) => (
+  <dl
+    className="pills-container"
+    key={members.map(member => `${member.translateContext}_${member.name}`).join(',')}
+  >
+    <dt />
+    <dd>
+      {members.map(member => (
+        <Pill
+          key={`${member.value[0].lat}_${member.value[0].lon}`}
+          color={templatesInfo[member.translateContext].color}
+        >
+          <Translate context={member.translateContext}>{member.label}</Translate>
+        </Pill>
+      ))}
+    </dd>
+  </dl>
+);
+
+const getMultiMemberInfo = (templatesInfo: TemplatesInfo) => (member: GroupMember) => (
+  <dl className="pills-container" key={`${member.translateContext}_${member.name}`}>
+    <dt>
+      <span>
+        <Translate context={member.translateContext}>{member.label}</Translate>
+        <> (</>
+        <Translate>linked</Translate>
+        <> </>
+        <Translate context={member.translateContext}>
+          {templatesInfo[member.translateContext].name}
+        </Translate>
+        <>)</>
```

Comment:
Maybe replacing the fragments with something like `<>
{' ('} 
<Translate>` could make this section more readable?

Reply:
I had trouble with eslint when using that syntax :/ but I agree it's much preferred. I'll give it a try again!

Reply:
I had to add an extra space after the closing parenthesis to work an eslint bug around, but I think it pays the price :)